mne/io/base.py

```diff
@@ -831,13 +848,80 @@ def get_data(self, picks=None, start=0, stop=None,
+                                 'one channel type with a unit '
+                                 f'{unit_ch_type}.')
+            units = {unit_ch_type[0]: units}  # make the str argument a dict
+
+        if units is None:
+            pass
+        elif isinstance(units, dict):
+            for ch_type, ch_unit in units.items():
+                # Check that the provided unit exists for the ch_type
+                unit_list = ch_unit.split('/')
+                si_unit_list = si_units_splitted[ch_type]
+                if len(unit_list) != len(si_unit_list):
+                    raise ValueError(f'{ch_unit} is not a valid unit.')
+                for i, unit in enumerate(unit_list):
+                    valid = [prefix + si_unit_list[i]
+                             for prefix in prefix_list]
+                    if unit not in valid:
+                        raise ValueError(f'{ch_unit} is not a valid unit.')
+
+                # Get the scaling factors
+                for i, unit in enumerate(unit_list):
+                    has_square = False
+                    # XXX power normally not used as csd cannot get_data()
+                    if unit[-1] == '²':
+                        has_square = True
+                    if unit == 'm':  # only the ECG unit starts like a prefix
+                        scaling = 1
+                    elif unit[0] in prefixes.keys():
+                        scaling = prefixes[unit[0]]
+                    else:
+                        scaling = 1
+                    if scaling != 1:
+                        if has_square:
+                            scaling *= scaling
+                        ch_types = self.get_channel_types(picks=picks)
+                        indices = [i_ch for i_ch, ch in enumerate(ch_types)
+                                   if ch == ch_type]
+                        if i == 0:
+                            ch_factors[indices] = ch_factors[indices] * scaling
+                        elif i == 1:
+                            ch_factors[indices] = ch_factors[indices] / scaling
+                        else:
+                            raise ValueError(f'{ch_unit} is not a valid unit.')
+        else:
+            raise TypeError('"units" must be None, str or dict, not '
+                            f'{type(units)}.')
+
         # convert to ints
         picks = np.atleast_1d(np.arange(self.info['nchan'])[picks])
         start = 0 if start is None else start
         stop = min(self.n_times if stop is None else stop, self.n_times)
         if len(self.annotations) == 0 or reject_by_annotation is None:
-            return self._getitem(
+            getitem = self._getitem(
                 (picks, slice(start, stop)), return_times=return_times)
+            if return_times:
+                data, times = getitem
+                return data * ch_factors[:, None], times
+            return getitem * ch_factors[:, None]
```

Comment:
`None` is valid, and works --> but `np.newaxis` is more explicit IMHO: https://stackoverflow.com/a/945313/5201771

```suggestion
                return data * ch_factors[:, np.newaxis], times
            return getitem * ch_factors[:, np.newaxis]
```